src/client.ts

```diff
@@ -40,35 +41,46 @@ export class InscriptionClient {
 
   /**
    * Constructs an instance of InscriptionClient.
-   * @param {string} key - The API key for authentication.
-   * @param {InscriptionEnv} environment - The environment for the client, either 'live' or 'dev'.
+   * @param {string} [key=''] - The API key for authentication.
+   * @param {InscriptionEnv} [environment='live'] - The environment (live or dev) (optional, defaults to live).
+   * @param {ClientOptions} [options] - Options for enabling L402 support.
    */
-  constructor(key: string = "", environment: InscriptionEnv = "live") {
+  constructor(key: string = "", environment: InscriptionEnv = "live", options?: ClientOptions) {
     this.api_key = key;
     this.env = environment;
 
+    /**
+     * Creates a new Axios instance with appropriate headers.
+     * @returns {AxiosInstance} The new Axios instance.
+     */
     const createInstance = (): AxiosInstance => {
+      const headers: Record<string, string> = {
+        Connection: "Keep-Alive",
+        "Content-Type": "application/json",
+      };
+
+      // Add the API key header only if this.api_key has a value
+      if (this.api_key) {
+        headers["x-api-key"] = this.api_key;
+      }
+
+      // Choose the base URL based on whether L402 is used or not
+      const baseURL = options?.useL402
+        ? "https://ordinalsbot.ln.sulu.sh"
+        : this.env === "live"
+          ? "https://api.ordinalsbot.com"
+          : "https://testnet-api.ordinalsbot.com";
+
+      // Create the Axios client with the appropriate base URL
       const client = axios.create({
-        baseURL:
-          environment === "live"
-            ? `https://api.ordinalsbot.com`
-            : `https://testnet-api.ordinalsbot.com`,
-        timeout: 30000,
-        headers: {
-          "x-api-key": this.api_key,
-          Connection: "Keep-Alive",
-          "Content-Type": "application/json",
-          "Keep-Alive": "timeout=10",
-          "User-Agent": packageVersion,
```

Comment:
Package version is missing from updated headers

Reply:
Done.